### PR TITLE
Move some util methods outside main.py

### PR DESCRIPTION
**Pull request by hao-fang**
The `main.py` gets quite big as we keep adding new pages & logics to it.
Ideally we can follow the "Large Applications" layout recommended [here](https://flask.palletsprojects.com/en/1.1.x/patterns/packages/). However, I don't have a good place to put the code logic for Freezer right now.
As an intermediate step, I move some utils methods from `main.py` to `miniconf/utils.py` and `miniconf/load_site_data.py`.

(It would be nice to follow the "Large Applications" layout in future.)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-PYTHON_FILES = main.py scripts/
+PYTHON_FILES = main.py miniconf/ scripts/ chat/
```
Comment (hao-fang) [inline Makefile:1]: also checks the `miniconf/` and `chat/` folders.

File: chat/make_poster_rooms.py [hunk 2]
```diff
@@ -1,51 +1,44 @@
 import argparse
 import csv
-import random
-import string
 
+import yaml
```
Comment (hao-fang) [inline chat/make_poster_rooms.py:4]: auto-formatted by `make format`

File: main.py [hunk 3]
```diff
@@ -1,41 +1,16 @@
...
-import yaml
 from flask import Flask, jsonify, redirect, render_template, send_from_directory
 from flask_frozen import Freezer
 from flaskext.markdown import Markdown
 
-site_data = {}
-by_uid = {}
-
-
-def main(site_data_path):
```
Comment (hao-fang) [inline main.py:17]: moved to `miniconf/load_site_data.py`.

File: main.py [hunk 4]
```diff
@@ -84,7 +57,7 @@ def papers():
 
 
 @app.route("/paper_vis.html")
-def paperVis():
```
Comment (hao-fang) [inline main.py:87]: drive-by

File: main.py [hunk 5]
```diff
@@ -110,74 +83,27 @@ def workshops():
...
-        list_fields[key] = extract_list_field(v, key)
-
-    return {
-        "id": v["UID"],
-        "title": v["title"],
-        "organizers": list_fields["authors"],
-        "abstract": v["abstract"],
-    }
-
-
```
Comment (hao-fang) [inline main.py:156]: moved to `miniconf/utils.py`

File: main.py [hunk 6]
```diff
@@ -110,74 +83,27 @@ def workshops():
...
 # ITEM PAGES
 
 
-@app.route("/poster_<poster>.html")
-def poster(poster):
-    uid = poster
+@app.route("/poster_<uid>.html")
+def poster(uid):
```
Comment (hao-fang) [inline main.py:90]: avoid the "redefined-outer-name" warning (same below)

**Comment by srush** [comment]
Sorry, I think I want to keep this as one main.py file. I like the simplicity. If we want to add some new feature that needs radically more code, we can add complexity. 

Can you split the rocketchat formatting changes to a separate PR?
